src/runtime/hstring.rs

```diff
@@ -171,32 +171,82 @@ impl From<&String> for HSTRING {
 
 impl PartialEq for HSTRING {
     fn eq(&self, other: &Self) -> bool {
-        self.as_wide() == other.as_wide()
+        *self.as_wide() == *other.as_wide()
     }
 }
 
 #[cfg(feature = "std")]
 impl PartialEq<String> for HSTRING {
     fn eq(&self, other: &String) -> bool {
-        self == other.as_str()
+        *self == **other
+    }
+}
+
+#[cfg(feature = "std")]
+impl PartialEq<String> for &HSTRING {
+    fn eq(&self, other: &String) -> bool {
+        **self == **other
+    }
+}
+
+#[cfg(feature = "std")]
+impl PartialEq<&String> for HSTRING {
+    fn eq(&self, other: &&String) -> bool {
+        *self == ***other
     }
 }
 
 impl PartialEq<str> for HSTRING {
     fn eq(&self, other: &str) -> bool {
-        self == other
+        self.as_wide().iter().copied().eq(other.encode_utf16())
+    }
+}
+
+impl PartialEq<str> for &HSTRING {
+    fn eq(&self, other: &str) -> bool {
+        **self == *other
     }
 }
 
 impl PartialEq<&str> for HSTRING {
     fn eq(&self, other: &&str) -> bool {
-        self.as_wide().iter().copied().eq(other.encode_utf16())
+        *self == **other
+    }
+}
+
+impl PartialEq<HSTRING> for str {
+    fn eq(&self, other: &HSTRING) -> bool {
+        *other == *self
     }
 }
 
 impl PartialEq<HSTRING> for &str {
     fn eq(&self, other: &HSTRING) -> bool {
-        other == self
+        *other == **self
+    }
+}
+
+impl PartialEq<&HSTRING> for str {
+    fn eq(&self, other: &&HSTRING) -> bool {
+        **other == *self
+    }
+}
+
+impl PartialEq<HSTRING> for String {
```

Comment:
Need to add `#[cfg(feature = "std")]` to these three impls.

Reply:
@kennykerr 
`String` is available from `alloc` so I have imported it from there. Is that ok?

Reply:
Hmm.. right now `String` is excluded when `std` is not requested in a lot of other places. How about we just apply `#[cfg(feature = "std")]` here for now and then I can re-enable `String` support consistently throughout as a subsequent change. Even with your change you'd have to also update the `for String` to `for ::alloc::string::String` to compile successfully... If you run `cargo check` locally you should pick that up.

Reply:
@kennykerr 
I've re-added the `#[cfg(feature = "std")` annotations back.